notebooks/Notebook3.md

```diff
@@ -2,33 +2,66 @@
 
 ## Description
 
-**TODO:** Fill in this part with information about your work this week:
-important design decisions, changes to previous decisions, open questions,
-exciting milestones, preliminary results, etc. Feel free to include images
-(e.g., a sketch of the design or a screenshot of a running program), links to
-code, and any other resources that you think will help clearly convey your
-design process.
+This week I wrote the parser for transitions and added a few features.
+For writing the transition parser, I ended up deciding to have a basic JavaScript file that has the functions all presentations need.
+Then separately, I can source the output of my transition parser which I decided to make a JSON list.
+Currently, I have it as a list with each element representing a transition and each transition containing a list the elements and effects to apply.
+I have all the transitions simplified into three types now: adding/removing classes, adding/removing transformations (translations, rotations, scaling, stretching), and changing element styles
+
+With both parsers working, I was able to replicate my presentation from last week into a [copy](<https://hmc-cs111-spring2023.github.io/project-code-occamkg/work%20files/testPres0/output>) produced by my program from the code.
+I also added a new [test presentation](<https://hmc-cs111-spring2023.github.io/project-code-occamkg/work%20files/testPres1/presentation>)!
+Some things I wanted to show with this are translating and scaling objects (images in this case), adding and removing classes (the blur added/removed from the text), and full size background images (the puppy!).
+The source input for this presentation can be found [here](<https://github.com/hmc-cs111-spring2023/project-code-occamkg/tree/master/work%20files/testPres1>) with the output in the `presentation` folder.
+I'm pretty satisfied with the transition syntax, but the element syntax is a bit bulky, so I may look into simplifying it further.
+I also added the slide number with some navigation controls, so that will be on every presentation.
 
 ## Questions
 
 **What is the most pressing issue for your project? What design decision do
 you need to make, what implementation issue are you trying to solve, or how
 are you evaluating your design and implementation?**
 
+Currently, whenever I want to render a change, I first have to save my file and then run the command `python path/to/program/run.py`.
+VS Code already has a very convenient feature (Open with Live Server) that rerenders HTML pages whenever you save anything in the workspace,
+but since my code doesn't output the HTML when it's saved, I can't take advantage of this.
+Ideally, what I would like is for the user to be able to choose/make a folder for their presentation and then have the code generate blank `.yaml`, `.css`, and `.frame` files.
+Then, whenever any files in this folder are saved, it would output the resulting web files.
```

Comment:
As a cheat, you might be able to use VS Code's "Build on save feature", and configure file types `yaml`, `css`, an `frame` to always call your compiler. 

If you set this as a global default, though, then `css` and `yaml` files in other projects might have unexpected behavior. There might be a way to restrict this setting to a VS Code "Profile", but I'm not sure.

Reply:
Browsing through your project, it reminded me of [Jekyll](https://jekyllrb.com/), which also [uses YAML](https://mmistakes.github.io/minimal-mistakes/docs/navigation/) among other things to generate static HTML content. Jekyll has [an option](https://github.com/jekyll/jekyll-watch) that will automatically rebuild when the source files change. To do this, you start a background Jekyll process that then relies on a Ruby gem called [listen](https://github.com/guard/listen) to monitor the files for changes. It all works pretty well with VSCode. Maybe there's something similar for Python?